src/tasks/minions/minigames/temporossActivity.ts

```diff
@@ -0,0 +1,104 @@
+import { increaseNumByPercent } from 'e';
+import { Task } from 'klasa';
+
+import { Emoji, Events } from '../../../lib/constants';
+import { getTemporossLoot } from '../../../lib/simulation/tempoross';
+import Fishing from '../../../lib/skilling/skills/fishing';
+import { SkillsEnum } from '../../../lib/skilling/types';
+import { ItemBank } from '../../../lib/types';
+import { TemporossActivityTaskOptions } from '../../../lib/types/minions';
+import { bankHasItem, channelIsSendable } from '../../../lib/util';
+import { handleTripFinish } from '../../../lib/util/handleTripFinish';
+import itemID from '../../../lib/util/itemID';
+
+export default class extends Task {
+	async run(data: TemporossActivityTaskOptions) {
+		const { userID, channelID, quantity, rewardBoost, duration } = data;
+		const user = await this.client.users.fetch(userID);
+		const currentLevel = user.skillLevel(SkillsEnum.Fishing);
+		const channel = await this.client.channels.fetch(channelID);
+
+		let loot: ItemBank = {};
+		let rewardTokens = quantity * 6;
+		if (rewardBoost > 0) {
+			rewardTokens = Math.ceil(increaseNumByPercent(rewardTokens, rewardBoost));
+		}
+
+		loot = getTemporossLoot(rewardTokens, currentLevel, user.bank()).bank;
+
+		if (bankHasItem(loot, itemID('Tiny tempor'))) {
+			this.client.emit(
+				Events.ServerNotification,
+				`${Emoji.TinyTempor} **${user.username}'s** minion, ${
+					user.minionName
+				}, just received a Tiny tempor! Their Tempoross KC is ${
+					(await user.getMinigameScore('Tempoross')) + quantity
```

Comment:
instead of grabbing the minigame score like this here, increment their minigame score near the top, and in the return, it will tell you what their new KC is.

```ts
const {newScore} = await incrementMinigameScore();

`Your KC is now ${newScore}".
```

However, keep in mind for your pet announcement, to make it more realistic, you might consider doing this, so its possible to get a 1-kc pet for example, if they do 10 kills of tempoross, it will always say they got the pet on the 10th KC. So, instead, you can just say a random kc between their old kc and their new kc, making it possible to "get" the pet on any KC, and not just in increments of how many they kill.

e.g.:

```js
const {previousScore, newScore} = increment();
const kcTheyGotPet = randInt(previousScore, newScore);
`You got the pet on your ${formatOrdinal(kcTheyGotPet} KC.`;
```